internal/engine/watchmanager.go

```diff
@@ -128,19 +130,23 @@ func (w *WatchManager) diff(ctx context.Context, st store.RStore) (setup []Watch
 	return setup, teardown
 }
 
-func dependenciesMatch(d1 []string, d2 []string) bool {
-	if len(d1) != len(d2) {
-		return false
-	}
-
-	for i, e1 := range d1 {
-		e2 := d2[i]
-		if e1 != e2 {
-			return false
+func watchRulesMatch(w1, w2 WatchableTarget) bool {
+	watchableTargetReposAsArrays := func(w WatchableTarget) ([]string, []string) {
+		var gitPaths, gitignores []string
+		for _, r := range w1.LocalRepos() {
+			gitPaths = append(gitPaths, r.LocalPath)
+			gitignores = append(gitignores, r.GitignoreContents)
 		}
+		return gitPaths, gitignores
 	}
 
-	return true
+	gitPaths1, gitignores1 := watchableTargetReposAsArrays(w1)
+	gitPaths2, gitignores2 := watchableTargetReposAsArrays(w2)
+
+	return sliceutils.StringSliceEquals(w1.Dependencies(), w2.Dependencies()) &&
```

Comment:
what about `func ignoresMatch(i1, i2 IgnorableTarget) bool {...}` and then
```
func watchesMatch(w1, w2 WatchableTarget) bool {
    return sliceutils.StringSliceEquals(w1.Dependencies(), w2.Dependencies()) &&
        ignoresMatch(i1, i2)
}
```
? then you can implement ignore-specific methods right on `IgnorableTarget` and don't need that awkward note above `WatchableTarget`

Reply:
I don't think that affects whether that awkward note is needed - if we add a field to `WatchableTarget`, we still need to update `watchesMatch` (and, likewise, if we add a field to `IgnorableTarget`, we need to update `ignoresMatch`).

I don't see a compelling argument one way or the other for whether ignoresMatch should be broken out. It feels kinda premature if it's not being called by anything else.

Reply:
fair nough